viewing_party/party.py

```diff
@@ -1,23 +1,189 @@
+        all_friends_movies.extend(user_data["friends"][i]["watched"])
+    
+    for movie in user_data["watched"]:
+        # if user movie title in friend's, continue?
+        if movie in all_friends_movies:
+            continue
+        # else (not in friends), append 
+        else:
+            user_unique_watched.append(movie)
+    return user_unique_watched
+    
+
+'''
+Return a list that determines which movies one of the friends watched but user didnt.
+'''
+
+def get_friends_unique_watched(user_data):
+
+    # get unique movies from friends
+    friends_unique_movies = []
+    # get movies from all friends 
+    all_friends_movies = []
+    for i in range (0, len(user_data["friends"])):
+        all_friends_movies.extend(user_data["friends"][i]["watched"])
+    for friend_movie in all_friends_movies:
+            # if friend movie title in user, continue
+            if friend_movie in user_data["watched"]:
+                continue
+        # else (not in user), append 
+            else:
+            #elif friend_movie not in user_data["friends"] and friends_unique_movies: ?? why doesnt work
+                if friend_movie not in friends_unique_movies:
+                    friends_unique_movies.append(friend_movie)
+    return friends_unique_movies    
+
 # ------------- WAVE 4 --------------------
-# -----------------------------------------
 
-# -----------------------------------------
+'''
+Get a list of recs_movie, on top of condition from friends_unique_movies in Wave 3, check if user has matching subscription.
+'''
+
+def get_available_recs(user_data):
+    recs_movie = []
+    friends_unique_movies = get_friends_unique_watched(user_data)
+    for i in range (len(friends_unique_movies)):
+        if friends_unique_movies[i]["host"] in user_data["subscriptions"]:
+            recs_movie.append(friends_unique_movies[i])        
+    return recs_movie       
+
 # ------------- WAVE 5 --------------------
-# -----------------------------------------
 
+'''
+Get a list of recs_movie, on top of condition from friends_unique_movies() in Wave 3, check if genre matches max genre using Wave 2.
+'''
+
+def get_new_rec_by_genre(user_data):
+    recs_movie = []
+    max_genre_movie = get_most_watched_genre(user_data)
+    friends_unique_movies = get_friends_unique_watched(user_data)
+    for i in range(len(friends_unique_movies)):
+        if friends_unique_movies[i]["genre"] == max_genre_movie:
+            recs_movie.append(friends_unique_movies[i]) 
+    return recs_movie
+
+'''
+Get a list of recs_movie, on top of condition from get_unique_watched() in Wave 3, check if movie in user favorites list.
+'''
+
+def get_rec_from_favorites(user_data):
+    recs_movie =[]
+    user_unique_watched = get_unique_watched(user_data)
+    for i in range(len(user_unique_watched)):
+        if user_unique_watched[i] in user_data["favorites"]:
+            recs_movie.append(user_unique_watched[i]) 
+    return recs_movie

```

Comment:
Nice job with wave 5! Great job calling helper functions! ✨ 

Another reminder to iterate directly over the values if you don't need to use the index of a list.